src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -12,19 +11,30 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int defaultBufferSize = 4096;
+        private const int DefaultBufferSize = 4096;
 
         private Stream _content;
+        private CancellationToken _cancellationToken;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, defaultBufferSize)
+            : this(content, DefaultBufferSize)
         {
         }
 
         public StreamContent(Stream content, int bufferSize)
+            : this(content, bufferSize, CancellationToken.None)
+        {
+        }
+
+        internal StreamContent(Stream content, CancellationToken cancellationToken)
```

Comment:
> We can defer the new overloads to a 4.2 System.Net.Http API (which I am already planning for) in the next few months.

To be clear, I wasn't suggesting we not try to do something for the 4.1 contract, just that we not hold up this PR on that.  It'd be good to at least sketch out what the new overloads would be and how we see them interplaying with existing code.  Depending on how its done, it could be problematic.  For example, let's say you have a method:
```
protected virtual Task Foo();
```
which the consumer calls.  Now we add a new overload:
```
protected virtual Task Foo(CancellationToken cancellationToken);
```
For the newly added method to be useful, we would need to switch to calling it instead of calling the former.  But no existing implementations will be overriding it, which means for compatibility the base implementation would need to delegate to the old implementation, at which point the CancellationToken is ignored.